app/Http/Controllers/GroupController.php

```diff
@@ -766,22 +790,55 @@ public function edit(Request $request, $id)
         ]);
     }
 
-    // TODO: is this alive?
     public function delete($id)
     {
-        if (FixometerHelper::hasRole($this->user, 'Administrator')) {
-            $r = $this->Group->delete($id);
+        $group = Group::where('idgroups', $id)->first();
+
+        $name = $group->name;
+
+        if (FixometerHelper::hasRole(Auth::user(), 'Administrator') && $group->canDelete()) {
```

Comment:
Not big deal here, but we have a method on user called hasRole which would be better to use in general.  I feel like FixometerHelper class in general should be deprecated over time.

Reply:
Thanks.  Changed.